src/Pester.Types.ps1

```diff
@@ -1,6 +1,15 @@
-if ($PSVersionTable.PSVersion.Major -ge 6) {
-    & $SafeCommands['Add-Type'] -Path "$PSScriptRoot/bin/netstandard2.0/Pester.dll"
-}
-else {
-    & $SafeCommands['Add-Type'] -Path "$PSScriptRoot/bin/net452/Pester.dll"
+# e.g. $minimumVersionRequired = "5.1.0.0" -as [version]
+$minimumVersionRequired = $ExecutionContext.SessionState.Module.PrivateData.RequiredAssemblyVersion -as [version]
```

Comment:
We probably need to be more clever than this here to make it work with pre-releases. I typically use beta1, beta2, rc1, rc2 etc... Those can be sorted by `sort` as long as we don't get to beta10 (or whatever pre-release, then it no longer sorts correctly).  I am okay with just Sort and throw on when suffix has more than 1 number. 

Or something like this:
```powershell
function Sort-Version ($Versions) { 
    $vs = foreach ($version in $Versions) {
        $v, $pre = $version -split "-",2

        $r = [PSCustomObject]@{
            String = $version
            Version = [Version] $v
            Char = $null
            Num = $null
        }

        if ($pre) { 
            if ($pre -match "(?<char>[a-z]+)(?<num>\d+)$") {
                $r.Char = $Matches.char
                $r.Num = [int] $Matches.num
            }
            else { 
                $r.Char = $pre
            }
        }

        if (-not $r.Num -and -not $r.Char) { 
            $r.Num = 1
            $r.Char = "z"
        }

        $r
    }

   $vs | Sort-Object -Property Version, Char, Num -Descending | Select-Object -ExpandProperty String
}

function Compare-Version($Left, $Right) {
    if ($Left -eq $Right) { 
        return 0
    }

    $sorted = Sort-Version $left, $right
    if ($sorted[0] -eq $left) { 
        return 1
    }
    else { 
        return -1
    }
}

Sort-Version "5.1.0", "5.1.0-beta", "5.1.0-beta1", "5.1.0-alpha1", "5.1.0-rc10", "5.1.0-rc2" , "5.1.0-rc1", "4.10.1"

Compare-Version "5.1.0" "5.1.0-rc1" # -> 1 left is higher
Compare-Version "5.1.0-rc10" "5.1.0-rc1" # -> 1 left is higher
Compare-Version "4.10.1" "5.1.0-alpha1" # -> -1 left is lower
```

Btw I just realized I forgot to set the dll version in 5.1.0.

Reply:
Pre-releases was ignored intentionally as they're work in progress and something you opt-in too install (with the possible issues that may follow). Don't have anything against including it though, so please add the commit if you'd like 🙂

Reply:
Yes. Can confirm that. Not using pre-releases is okay.